File: apps/worker-notification/channels/webhook.ts
import type { NotificationChannel, NotificationPayload } from "./types";

export class WebhookChannel implements NotificationChannel {
    async send(payload: NotificationPayload): Promise<void> {
        console.log(`Webhook ${payload.websiteId} is ${payload.eventType} at ${payload.occurredAt}`)
    }
}

File: apps/api-go/internal/channels/webhook.go
package channels

import (
	"context"
	"log"
)

type WebhookChannel struct {
}

func NewWebhookChannel() *WebhookChannel {
	return &WebhookChannel{}
}

func (wc *WebhookChannel) Send(
	ctx context.Context,
	payload NotificationPayload,
) (string, error) {
	log.Printf(
		"Website %s is %s at %s",
		payload.WebsiteID,
		payload.EventType,
		payload.OccurredAt,
	)

	return "ok", nil
}
